vsm

```diff
@@ -65,12 +66,15 @@ config_tree = None
 # parsed code so we can't encapsulate it
 node_refs = {}
 state = None
+signal_to_num = {}
 
 
 def _format_signal_msg(signal, value, indicator):
-    # NOTE: the "[SIGNUM]" field is a temporary placeholder for the signal ID
-    return '{} {},{},[SIGNUM],{}'.format(indicator, get_runtime(),
-                                         signal, repr(value))
+    signum = "[SIGNUM]"
+    if signal in signal_to_num:
+        signum = str(signal_to_num[signal])
```

Comment:
Fixed. I was over-compensating for the fact that Python fails to do this automatically in most other cases.